prvocisla.cpp

```diff
@@ -29,16 +29,19 @@ int * sieve(int n){
 
     for (int p = 2; p<= n; p++){
         if (A[p] != 0) numPrime++;
+    
     }
 
     L = (int *) malloc(numPrime*sizeof(int));
 
     int j = 0;
+    
     while (j < numPrime) {
-         for (int p = 2; p<= n; p++){
-             if (A[p] != 0) {
-                 L[j] = A[p];
-                 j++;
+        for (int p = 2; p<= n; p++){
```

Comment:
Abych klarifikoval, ten koment nad timto ^:

toto je ta druha smycka; ktera si myslim ze by mela jako jedina stacit (tu nadtim=prvni bych nahradil assertem ;)) ...

Reply:
nn. nechapes ten algoritmus jaky ma prubeh.

Reply:
> nn. nechapes ten algoritmus jaky ma prubeh.

Enlighten me... ("nechapes" = asi to potrebujes vysvetlit lepe nez jen toto;  bych  ti taky misto tech komentaru mohl napsat jen "je to blbe" .... jak bys mi pak asi byla vdecna?)

toto je pokud vim uz jen kopirovani pole.... pochop ze narazim jen na to co se deje na radku 37 a nize....

Reply:
_uz jsem videl hodne logickych konstrukci (patterns = casto se opakuji); a jak pisu, proste toto konkretne vypada podivne - tj. idealni na refactoring..._